tests/unit/test_workspace.py

```diff
@@ -1,88 +1,72 @@
 import os
 
+import pytest
 import numpy as np
 
 from jina.executors import BaseExecutor
-from tests import JinaTestCase
 
-cur_dir = os.path.dirname(os.path.abspath(__file__))
+@pytest.mark.parametrize('replica_id', [1,2,3])
+def test_share_workspace(tmpdir, replica_id):
```

Comment:
The term replica_id should be replaces by pea_id.

Reply:
this is what we have in the definition:
```python
    @classmethod
    def load_config(cls: Type[AnyExecutor], source: Union[str, TextIO], separated_workspace: bool = False,
                    replica_id: int = 0) -> AnyExecutor:
```

Reply:
I see, we will refactor then

Reply:
do you have idea that I always get an error: `E           AttributeError: 'NoneType' object has no attribute 'header'` when calling `add`, such as
```python
def test_compound_indexer_rw():
    all_vecs = np.random.random([6, 5])
    for j in range(3):
        with BaseExecutor.load_config('yaml/test-compound-indexer2.yml', True, j) as a:
            assert a[0] == a['test_meta']
            assert not a[0].is_updated
            assert not a.is_updated
            a[0].add([j, j * 2, j * 3], [bytes(j), bytes(j * 2), bytes(j * 3)])
```
The last test case never passed locally

Reply:
wired.. passed in CI..

Reply:
Did u clean ur workspace?